### PR TITLE
added a new command for getting device date and time

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
@Jonahss Do we have any method existing to get current device id for android? Also please review.

### CODE REVIEW

**Comment by Jonahss** [comment]
Ah, you can use `this.adb`. It's a reference to an ADB object from [appium-adb](https://github.com/appium/appium-adb). By the time the session is created, it already has the udid set on the adb object and it adds the udid arguments automatically.

**Comment by SrinivasanTarget** [comment]
@Jonahss Is this looks fine?

File: lib/commands/general.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { fs } from 'appium-support';
 import B from 'bluebird';
 //import { errors } from 'mobile-json-wire-protocol';
 import log from '../logger';
+import { exec } from 'teen_process';
```
Comment (imurchie) [inline lib/commands/general.js]: No need for this.

File: lib/commands/general.js [hunk 2]
```diff
@@ -25,6 +26,16 @@ commands.keys = async function (keys) {
   return await this.bootstrap.sendAction('setText', params);
 };
 
+commands.getDeviceTime = async function() {
+  log.info('Attempting to capture android device date and time');
+  try {
+    await this.adb.shell(['-s', this.opts.udid],'date')
```
Comment (imurchie) [inline lib/commands/general.js]: Need to get the standard output from this call, to use in the next line. Something like we find here https://github.com/appium/appium-android-driver/blob/6e6478c6acab054b1966fdd2090412090726f3ff/lib/webview-helpers.js#L20

Comment (imurchie) [inline lib/commands/general.js]: Also need a semicolon at the end.

**Comment by imurchie** [comment]
A few comments to address the CI failure.

File: lib/commands/general.js [hunk 3]
```diff
@@ -25,6 +25,16 @@ commands.keys = async function (keys) {
   return await this.bootstrap.sendAction('setText', params);
 };
 
+commands.getDeviceTime = async function() {
+  log.info('Attempting to capture android device date and time');
+  try {
+    let out = await this.adb.shell(['-s', this.opts.udid],'date');
```
Comment (imurchie) [inline lib/commands/general.js]: Can you add a space before `'date'`? We are then on-style.

**Comment by imurchie** [comment]
One more quick change, then squash and we can commit! Thanks for this!

**Comment by SrinivasanTarget** [comment]
Yup Its squashed @imurchie

**Comment by imurchie** [comment]
Thanks for this!

**Pull request merged**